controllers/gslb_controller_reconciliation.go

```diff
@@ -176,7 +141,74 @@ func (r *GslbReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 		m.IncrementError(gslb)
 		return result.RequeueError(fmt.Errorf("getting GSLB servers (%s)", err))
 	}
-	gslb.Status.Servers = servers
+
+	var zone utils.DNSZone
+	var matchedServers []*k8gbv1beta1.Server
+	for _, server := range servers {
+		log.Debug().
+			Str("server", server.Host).
+			Msg("Checking server")
+		for _, z := range r.Config.DNSZones {
+			log.Debug().
+				Str("zone", z.Zone).
+				Msg("Checking zone")
+			if strings.HasSuffix(server.Host, z.Zone) {
+				log.Debug().
+					Str("server", server.Host).
+					Str("zone", z.Zone).
+					Msg("Matched zone to server host")
+				zone = z
+				matchedServers = append(matchedServers, server)
+			}
+		}
+	}
+
+	if zone == (utils.DNSZone{}) {
```

Comment:
Is this logic to trigger a requeue if no server matched necessary?
Let's suppose the configured dnsZone is "example.com". There is a GSLB resource with a server that has the hostname "cloud.example.com". Then, the users swaps the hostname to "cloud.other.com". With this implementation the GSLB will never stop advertising "cloud.example.com".

Reply:
I will test this out 👍